setup.py

```diff
@@ -161,7 +167,8 @@ def run(self):
             [
                 configure, "--disable-shared", "--enable-static",
                 "--disable-debug", "--disable-dependency-tracking",
-                "--with-pic", "--prefix", os.path.abspath(self.build_clib),
+                "--disable-ssp", "--with-pic",
```

Comment:
Passing `--disable-ssp` unconditionally removes a security feature that works in most contexts. If we need to disable it on some platforms we need to be smarter about how we detect support.

Reply:
I investigated this a little further and found that the issue with SSP is caused by non-static linking of this library into libsodium. Since libsodium uses libssp (unless --disable-ssp is specified) but doesn't statically link to it on Solaris, we have to add dynamic linking to this library in pynacl itself (`-lssp`, or, add `ssp` to `libraries` in `bindings/build.py`). However, we can't be sure that libssp is really used (e.g. if the package, CSWlibssp0, is not installed - libsodium's `configure` checks SSP's availability), so we don't want to blindly add libssp to the linker. Which brings me back to the simple solution of just using `--disable-ssp` on Solaris.

I saw a few comments that see this as a bug on libsodium's side (and even a forks that fixes this in their configure script - https://git.luxing.im/hlx98007/shadowsocksr-libev/commit/5d83a5ee7fb8afb883ff51a3e75e0e0a5affefd9) but I don't know if libsodium will accept this upstream.

Do you think it's ok to add checks specifically for Solaris in setup.py to add this flag if we detect this operating system?